gmso/lib/potential_templates.py

```diff
@@ -52,7 +48,7 @@ def _load_template_json(item, json_dir=JSON_DIR):
 class PotentialTemplate(AbstractPotential):
     """Template for potential objects to be re-used."""
 
-    expected_parameters_dimensions_: Dict[str, sympy.Expr] = Field(
+    expected_parameters_dimensions: Dict[str, sympy.Expr] = Field(
```

Comment:
What's the reason this was changed from `expected_parameters_dimensions_` to `expected_parameters_dimensions`

Reply:
good catch. I converted all these to remove the `_` at the end, thinking that the new pydantic has an updated way to handle custom setter/getter. They don't, so I have to revert back to the old way, having a name with `_` at the end and an `alias` for each. I think I just forgot to revert this.

Reply:
ah okay, then I'll let you handle that. It looks like the same issue pops up with the restraints in atom.py when I tried to merge them, so I can let you revert whatever commit removed the `_` and add the getter and setters.